rcsb/workflow/bcif/task_functions.py

```diff
@@ -0,0 +1,370 @@
+    ]
+    return tasks
+
+
+def batchTask(
+    tasks,
+    remotePath,
+    updateBase,
+    outfileSuffix,
+    outputContentType,
+    outputHash,
+    contentType,
+    dictionaryApi,
+    temppath,
+    maxTempFiles,
+):
+    logger.info("processing %d tasks", len(tasks))
+    for task in tasks:
+        singleTask(
+            task,
+            remotePath,
+            updateBase,
+            outfileSuffix,
+            outputContentType,
+            outputHash,
+            contentType,
+            dictionaryApi,
+            temppath,
+            maxTempFiles,
+        )
+
+
+def singleTask(
+    pdbId,
+    remotePath,
+    updateBase,
+    outfileSuffix,
+    outputContentType,
+    outputHash,
+    contentType,
+    dictionaryApi,
+    temppath,
+    maxTempFiles,
+    counter=[0],
+):
+    if contentType in ["pdb", "ihm"]:
+        pdbId = pdbId.lower()
+    elif contentType == "csm":
+        pdbId = pdbId.upper()
+    remoteFileName = "%s%s" % (
+        pdbId,
+        outfileSuffix.replace(".bcif.gz", ".cif.gz").replace(".bcif", ".cif"),
+    )
+
+    # form input cifFilePath
+    if not remotePath.startswith("http"):
+        # local file
+        cifFilePath = os.path.join(remotePath, remoteFileName)
+        if not os.path.exists(cifFilePath):
+            logger.warning("%s not found", cifFilePath)
+            return
+    else:
+        cifFilePath = os.path.join(remotePath, pdbId[-3:-1], remoteFileName)
+        if contentType == "csm":
+            cifFilePath = os.path.join(
+                remotePath, pdbId[0:2], pdbId[-6:-4], pdbId[-4:-2], remoteFileName
+            )
+        elif contentType == "ihm":
+            cifFilePath = os.path.join(
+                remotePath, pdbId[-3:-1], pdbId, "structures", remoteFileName
+            )
+
+    # form output bcifFilePath
+    bcifFilePath = getBcifFilePath(
+        pdbId, outfileSuffix, updateBase, contentType, outputContentType, outputHash
+    )
+    if not bcifFilePath:
+        raise ValueError("failed to form bcif file path")
+    if os.path.exists(bcifFilePath):
+        # earlier timestamp ... overwrite
+        os.unlink(bcifFilePath)
+        if os.path.exists(bcifFilePath):
+            raise PermissionError("file %s not removed" % bcifFilePath)
+
+    # make nested directories
+    dirs = os.path.dirname(bcifFilePath)
+    if not os.path.exists(dirs):
+        os.makedirs(dirs)
+
+    # convert to bcif
+    result = convert(cifFilePath, bcifFilePath, temppath, dictionaryApi)
+    if not result:
+        raise Exception("failed to convert %s" % cifFilePath)
+
+    counter[0] += 1
+
+    # remove temp files
+    if counter[0] >= maxTempFiles:
+        removeTempFiles(tempPath=temppath)
+        counter[0] = 0
+
+
+def getBcifFilePath(
+    pdbId, outfileSuffix, updateBase, contentType, outputContentType, outputHash
+):
+    bcifFileName = "%s%s" % (pdbId, outfileSuffix)
+    bcifFilePath = None
+    if contentType == "pdb":
```

Comment:
Don't we have a content type enum somewhere already @piehld @trumbullm ? That would be the ideal why to handle this to have more strong typing.